controller/value_resolver.rst

```diff
@@ -168,132 +168,88 @@ PSR-7 Objects Resolver:
-type-hinted method arguments:
+.. versionadded:: 6.2
 
-.. configuration-block::
-
-    .. code-block:: yaml
-
-        # config/packages/sensio_framework_extra.yaml
-        sensio_framework_extra:
-            request:
-                converters: true
-                auto_convert: false
-
-    .. code-block:: xml
-
-        <!-- config/packages/sensio_framework_extra.xml -->
-        <?xml version="1.0" encoding="UTF-8" ?>
-        <container xmlns="http://symfony.com/schema/dic/services"
-            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-            xmlns:sensio-framework-extra="http://symfony.com/schema/dic/symfony_extra"
-            xsi:schemaLocation="http://symfony.com/schema/dic/services
-                https://symfony.com/schema/dic/services/services-1.0.xsd
-                http://symfony.com/schema/dic/symfony_extra
-                https://symfony.com/schema/dic/symfony_extra/symfony_extra-1.0.xsd">
-
-            <sensio-framework-extra:config>
-                <request converters="true" auto-convert="false"/>
-            </sensio-framework-extra:config>
-        </container>
-
-    .. code-block:: php
-
-        // config/packages/sensio_framework_extra.php
-        $container->loadFromExtension('sensio_framework_extra', [
-            'request' => [
-                'converters' => true,
-                'auto_convert' => false,
-            ],
-        ]);
+    The ``ValueResolverInterface`` was introduced in Symfony 6.2. Prior to
+    6.2, you had to use the
+    :class:`Symfony\\Component\\HttpKernel\\Controller\\ArgumentValueResolverInterface`,
+    which defines different methods.
 
 Adding a new value resolver requires creating a class that implements
-:class:`Symfony\\Component\\HttpKernel\\Controller\\ArgumentValueResolverInterface`
-and defining a service for it. The interface defines two methods:
-
-``supports()``
-    This method is used to check whether the value resolver supports the
-    given argument. ``resolve()`` will only be called when this returns ``true``.
-``resolve()``
-    This method will resolve the actual value for the argument. Once the value
-    is resolved, you must `yield`_ the value to the ``ArgumentResolver``.
+:class:`Symfony\\Component\\HttpKernel\\Controller\\ValueResolverInterface`
+and defining a service for it.
 
-Both methods get the ``Request`` object, which is the current request, and an
+This interface contains a ``resolve()`` method, which is called for each
+argument of the controller. It receives the current ``Request`` object and an
 :class:`Symfony\\Component\\HttpKernel\\ControllerMetadata\\ArgumentMetadata`
-instance. This object contains all information retrieved from the method signature
-for the current argument.
+instance, which contains all information from the method signature. The
+method should return either an empty array (if it cannot resolve this
+argument) or an array with the resolved value(s).
```

Comment:
Why does it return always an array? What happens when you return more than a value in the array? Which one is picked? Thanks.

Reply:
The array is used for variadic arguments. You'll get a logic exception if you return more than one item and the argument isn't variadic

Reply:
Ah, it makes sense now. Thanks.